src/app.rs

```diff
@@ -1322,7 +1322,8 @@ impl Config {
         }
     }
 
-    fn request_tiles(&mut self, request_interval: Interval) -> Vec<TileID> {
+    fn request_tiles(&mut self, cx: &mut Context) -> Vec<TileID> {
+        let request_interval = cx.view_interval.intersection(cx.total_interval);
```

Comment:
[cb0343a](https://github.com/StanfordLegion/prof-viewer/pull/40/commits/cb0343a2c3e3c1fe08c3efd144e54084101e032d)

BTW the viewer still crashes if you scroll far enough off the screen that all the data is outside the view interval. But that points towards being defensive on the other end.

Reply:
Yeah. We should probably harden the `StateDataSource` to avoid crashing when the interval is empty. There are (at least in principle) reasons why you might want to do this when viewing multiple profiles. (E.g., you could be off the end of one profile but not of another.)

Reply:
I can submit a PR to legion next week after this one is merged